src/deployment.js

```diff
@@ -7,181 +7,183 @@ const axios = require('axios')
+      core.setOutput('page_url', this.deploymentInfo != null ? this.deploymentInfo['page_url'] : '')
+      const timeout = Number(core.getInput('timeout'))
+      const reportingInterval = Number(core.getInput('reporting_interval'))
+      const maxErrorCount = Number(core.getInput('error_count'))
+      var startTime = Date.now()
+      var errorCount = 0
+
+      // Time in milliseconds between two deployment status report when status errored, default 0.
+      var errorReportingInterval = 0
+
+      /*eslint no-constant-condition: ["error", { "checkLoops": false }]*/
+      while (true) {
+        // Handle reporting interval
+        await new Promise(r => setTimeout(r, reportingInterval + errorReportingInterval))
+
+        // Check status
+        var res = await axios.get(statusUrl, {
+          headers: {
+            Authorization: `token ${this.githubToken}`
           }
+        })
 
-          if (res.status != 200 || !!errorStatus[res.data.status]) {
-            errorCount++
+        if (res.data.status == 'succeed') {
+          core.info('Reported success!')
+          core.setOutput('status', 'succeed')
+          break
+        } else if (res.data.status == 'deployment_failed') {
+          // Fall into permanent error, it may be caused by ongoing incident or malicious deployment content or exhausted automatic retry times.
+          core.setFailed('Deployment failed, try again later.')
+          break
+        } else if (res.data.status == 'deployment_content_failed') {
+          // The uploaded artifact is invalid.
+          core.setFailed(
+            'Artifact could not be deployed. Please ensure the content does not contain any hard links, symlinks and total size is less than 10GB.'
+          )
+          break
+        } else if (errorStatus[res.data.status]) {
+          // A temporary error happened, will query the status again
+          core.info(errorStatus[res.data.status])
+        } else {
+          core.info('Current status: ' + res.data.status)
+        }
 
-            // set the Maximum error reporting interval greater than 15 sec but below 30 sec.
-            if (errorReportingInterval < 1000 * 15) {
-              errorReportingInterval = errorReportingInterval << 1 | 1
-            }
-          } else {
-            // reset the error reporting interval once get the proper status back.
-            errorReportingInterval = 0
-          }
+        if (res.status != 200 || !!errorStatus[res.data.status]) {
+          errorCount++
 
-          if (errorCount >= maxErrorCount) {
-            core.info('Too many errors, aborting!')
-            core.setFailed('Failed with status code: ' + res.status)
-            break
+          // set the Maximum error reporting interval greater than 15 sec but below 30 sec.
+          if (errorReportingInterval < 1000 * 15) {
+            errorReportingInterval = (errorReportingInterval << 1) | 1
```

Comment:
Unrelated to these formatting changes but I want to call it out since I see it here: why are we using bitwise operators for calculating an interval? 😅

Reply:
That's just fancy for `* 2`. Did not touch that yet but I don't like it. I may make another pass later there.